src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -6052,13 +6053,52 @@ TEST_F(storage_test_fixture, find_sliding_ranges) {
 	                                                             // 6   - (Active)
 	  .new_start_offset=model::offset{100},
 	  .expected_ranges = {{3, 6}}},
+      sliding_ranges_test_case{
+	  .desc="All mergeable segments, but number of max segments is limited to 0 via cluster config.",
+	  .segment_fields={
+	    {1_KiB, model::term_id{0}, true},  // 0 -
+	    {1_KiB, model::term_id{0}, true},  // 1 -
+	    {1_KiB, model::term_id{0}, true},  // 2 -
+	    {1_KiB, model::term_id{0}, true},  // 3 -
+	    {1_KiB, model::term_id{0}, true},  // 4 -
+	    {1_KiB, model::term_id{0}, true}}, // 5 -
+	                                       // 6 - (Active)
+	  .max_segment_count = 0,
+	  .expected_ranges = {}},
+      sliding_ranges_test_case{
+	  .desc="All mergeable segments, but number of max segments is limited to 1 via cluster config.",
+	  .segment_fields={
+	    {1_KiB, model::term_id{0}, true},  // 0 -
+	    {1_KiB, model::term_id{0}, true},  // 1 -
+	    {1_KiB, model::term_id{0}, true},  // 2 -
+	    {1_KiB, model::term_id{0}, true},  // 3 -
+	    {1_KiB, model::term_id{0}, true},  // 4 -
+	    {1_KiB, model::term_id{0}, true}}, // 5 -
+	                                       // 6 - (Active)
+	  .max_segment_count = 1,
+	  .expected_ranges = {}},
+      sliding_ranges_test_case{
+	  .desc="All mergeable segments, but number of max segments is limited to 2 via cluster config.",
+	  .segment_fields={
+	    {1_KiB, model::term_id{0}, true},  // 0 A
+	    {1_KiB, model::term_id{0}, true},  // 1 A
+	    {1_KiB, model::term_id{0}, true},  // 2  B
+	    {1_KiB, model::term_id{0}, true},  // 3  B
+	    {1_KiB, model::term_id{0}, true},  // 4   C
+	    {1_KiB, model::term_id{0}, true}}, // 5   C
+	                                       // 6    - (Active)
+	  .max_segment_count = 2,
+	  .expected_ranges = {{0, 2}, {2, 4}, {4, 6}}},
     };
     // clang-format on
 
     for (int test_case_index = 0; const auto& test_case : test_cases) {
         vlog(e2e_test_log.info, "Running test case: {}", test_case.desc);
         const auto& segment_fields = test_case.segment_fields;
         const auto& expected_ranges = test_case.expected_ranges;
+        config::shard_local_cfg()
```

Comment:
I'm not sure with the new gtest and bazel test runs, but just a heads up, it's worth checking if this pollutes subsequent tests, and if so, using a scoped config to reset this between tests